### ISSUE TITLE
How to specify the instances for batches

### ISSUE THREAD

**Issue by nrjvarshney** [pytorch/pytorch#47118]
I am trying to solve a multi-task learning problem where I want to implement a homogeneous epoch sampling strategy (i.e in a single batch, instances from only one task are present and such batches are shuffled).

For example, Bij represents ith batch during training is of jth task
Let's assume tasks are A,B,C
B1A, B2B, B3A, B4C, B5B, ....
So a batch contains instances of one task only.

How can this be achieved?

**Comment by SsnL** [comment]
Use a custom batch_sampler for your dataloader. 

Use forums for questions. Github is for bugs and issues.
